src/Microsoft.AspNetCore.Mvc.Razor/Internal/FileProviderRazorProjectItem.cs

```diff
@@ -27,11 +30,50 @@ public FileProviderRazorProjectItem(IFileInfo fileInfo, string basePath, string
 
         public override string PhysicalPath => FileInfo.PhysicalPath;
 
-        public override string RelativePhysicalPath { get; }
+        public override string RelativePhysicalPath
+        {
+            get
+            {
+                if (_relativePhysicalPath == null)
+                {
+                    if (Exists)
+                    {
+                        if (_root != null && !string.IsNullOrEmpty(PhysicalPath) && PhysicalPath.StartsWith(_root))
```

Comment:
`StartsWith(_root, StringComparison.OrdinalIgnoreCase)`

Reply:
Is the _root guaranteed to be normalized i.e. ending in a directory separator? Consider:

```
_root = c:\Users\Ajay\Test1
PhysicalPath = c:\Users\Ajay\Test123\File.txt
```